Add support for multiple ZIM langs + validate Language metadata

benoit74 opened the pull request:
Fix #300 

Changes:
- isolate method to parse language received either from input (CLI) args or automatically retrieved from HTML metadata
- add unit testing on this method
- add support for a list of languages, permissive to spaces which will be trimmed and preserving list of languages order
- add validation of language metadata, including early validation if the lang comes from input (CLI) args
- add a real fallback to `eng` as indicated in usage, wasn't done when no lang was provided at CLI and no lang was found in WARC records

rgaudin commented on src/warc2zim/language.py:
```diff
@@ -0,0 +1,28 @@
+                langs.append(parsed_lang)
+        except Exception:
+            logger.error(f"Invalid language setting `{lang}`. Using `eng`.")
+            return "eng"
```
Looks like a `fra,unkown` would return `eng` while I would expect `fra`.
Shouldn't we log and skip un-parsable ones instead?

benoit74 replied:
You're right, this is silly.
Let's log and skip un-parsable ones.

benoit74 commented:
I've also changed log level to WARNING since these problems are recovered, and I added a WARNING log when the general fallback is applied (when no language has been specified at input - CLI - and no language has been found in main entry)

rgaudin reviewed (approved):
👍
